DataFormats/TrackSoA/test/TestWriteHostTrackSoA.cc

```diff
@@ -0,0 +1,52 @@
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/EDPutToken.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/TrackSoA/interface/TracksHost.h"
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+namespace edmtest {
+
+  class TestWriteHostTrackSoA : public edm::global::EDProducer<> {
+  public:
+    TestWriteHostTrackSoA(edm::ParameterSet const&);
+    void produce(edm::StreamID, edm::Event&, edm::EventSetup const&) const override;
+    static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+    using TracksOnHost = ::reco::TracksHost;
+
+  private:
+    unsigned int trackSize_;
+    edm::EDPutTokenT<TracksOnHost> putToken_;
+  };
+
+  TestWriteHostTrackSoA::TestWriteHostTrackSoA(edm::ParameterSet const& iPSet)
+      : trackSize_(iPSet.getParameter<unsigned int>("trackSize")), putToken_(produces()) {}
+
+  void TestWriteHostTrackSoA::produce(edm::StreamID, edm::Event& iEvent, edm::EventSetup const&) const {
+    TracksOnHost tracks({{int(trackSize_), int(4 * trackSize_)}}, cms::alpakatools::host());
```

Comment:
Just curious: it doesn't compile without the explicit casts to `int` ?

Reply:
Yes it complains with 

```bash
src/DataFormats/TrackSoA/test/TestWriteHostTrackSoA.cc:34:33: error: narrowing conversion of '(unsigned int)((const edmtest::TestWriteHostTrackSoA*)this)->edmtest::TestWriteHostTrackSoA::trackSize_' from 'unsigned int' to 'int' [-Werror=narrowing]
   34 |     ::reco::TracksHost tracks({{trackSize_, int(4 * trackSize_)}}, cms::alpakatools::host());
 ```

Reply:
Ah, now we have those enabled as well :(